File: android/src/main/java/com/amplitude/AmplitudeAnalytics.java
package com.amplitude;


import com.amplitude.api.Amplitude;
import com.amplitude.api.Identify;
import com.amplitude.api.TrackingOptions;
import com.facebook.react.bridge.ReactApplicationContext;
import com.facebook.react.bridge.ReactContextBaseJavaModule;
import com.facebook.react.bridge.ReactMethod;
import com.facebook.react.bridge.ReadableArray;
import com.facebook.react.bridge.ReadableMap;

import org.json.JSONException;
import org.json.JSONObject;

import java.util.ArrayList;

public class AmplitudeAnalytics extends ReactContextBaseJavaModule  {

    public static final String REACT_CLASS = "AmplitudeAnalytics";

    public AmplitudeAnalytics(ReactApplicationContext context) {
        super(context);
    }


    @ReactMethod void initialize(String apiKey){
        Amplitude.getInstance().initialize(getReactApplicationContext().getApplicationContext(), apiKey)
                .enableForegroundTracking(getCurrentActivity().getApplication());
    }

    @ReactMethod void setTrackingOptions(ReadableArray options) {
        TrackingOptions trackingOptions = new TrackingOptions();
        ArrayList optionsList = options.toArrayList();
        if (optionsList.contains("disableAdid")){
            trackingOptions.disableAdid();
        }
        if (optionsList.contains("disableCarrier")){
            trackingOptions.disableCarrier();
        }
        if (optionsList.contains("disableCity")){
            trackingOptions.disableCity();
        }
        if (optionsList.contains("disableCountry")){
            trackingOptions.disableCountry();
        }
        if (optionsList.contains("disableDeviceBrand")){
            trackingOptions.disableDeviceBrand();
        }
        if (optionsList.contains("disableDeviceManufacturer")){
            trackingOptions.disableDeviceManufacturer();
        }
        if (optionsList.contains("disableDeviceModel")){
            trackingOptions.disableDeviceModel();
        }
        if (optionsList.contains("disableDMA")){
            trackingOptions.disableDma();
        }
        if (optionsList.contains("disableIPAddress")){
            trackingOptions.disableIpAddress();
        }
        if (optionsList.contains("disableLanguage")){
            trackingOptions.disableLanguage();
        }
        if (optionsList.contains("disableLatLng")){
            trackingOptions.disableLatLng();
        }
        if (optionsList.contains("disableOSName")){
            trackingOptions.disableOsName();
        }
        if (optionsList.contains("disableOSVersion")){
            trackingOptions.disableOsVersion();
        }
        if (optionsList.contains("disablePlatform")){
            trackingOptions.disablePlatform();
        }
        if (optionsList.contains("disableRegion")){
            trackingOptions.disableRegion();
        }
        if (optionsList.contains("disableVersionName")){
            trackingOptions.disableVersionName();
        }
        Amplitude.getInstance().setTrackingOptions(trackingOptions);
    }

    @ReactMethod void setUserPropertyForOnce(String key, String value){
        Identify identify = new Identify().setOnce(key, value);
        Amplitude.getInstance().identify(identify);
    }

    @ReactMethod void unSetUserProperty(String key){
        Identify identify = new Identify().unset(key);
        Amplitude.getInstance().identify(identify);
    }

    @ReactMethod void setDeviceId(String deviceId){
        Amplitude.getInstance().setDeviceId(deviceId);
    }

    @ReactMethod
    public void logEvent(String eventName, ReadableMap parameters) throws JSONException {

        JSONObject eventProperties = Utils.convertMapToJson(parameters);
        com.amplitude.api.Amplitude.getInstance().logEvent(eventName, eventProperties);
    }



    @Override
    public String getName() {
        return REACT_CLASS;
    }
}

File: ios/AmplitudeAnalytics/AmplitudeAnalytics.m
//  Created by react-native-create-bridge

#import <UIKit/UIKit.h>
#import "AmplitudeAnalytics.h"
// import RCTBridge
#if __has_include(<React/RCTBridge.h>)
#import <React/RCTBridge.h>
#elif __has_include(“RCTBridge.h”)
#import “RCTBridge.h”
#else
#import “React/RCTBridge.h” // Required when used as a Pod in a Swift project
#endif

// import RCTEventDispatcher
#if __has_include(<React/RCTEventDispatcher.h>)
#import <React/RCTEventDispatcher.h>
#elif __has_include(“RCTEventDispatcher.h”)
#import “RCTEventDispatcher.h”
#else
#import “React/RCTEventDispatcher.h” // Required when used as a Pod in a Swift project
#endif
#import "Amplitude.h"
#import "AMPTrackingOptions.h"

@interface AmplitudeAnalytics ()
{
    
}
@end
@implementation AmplitudeAnalytics
@synthesize bridge = _bridge;

// Export a native module
// https://facebook.github.io/react-native/docs/native-modules-ios.html
RCT_EXPORT_MODULE();

// Export methods to a native module
// https://facebook.github.io/react-native/docs/native-modules-ios.html

RCT_EXPORT_METHOD(initialize:(NSString *)apiKey){
    [[Amplitude instance] initializeApiKey:apiKey];
}

RCT_EXPORT_METHOD(setTrackingOptions:(NSArray *)options){
    AMPTrackingOptions *trackingOptions = [AMPTrackingOptions options];
    if([options containsObject:@"disableCarrier"]){
        [trackingOptions disableCarrier];
    }
    if([options containsObject:@"disableCity"]){
        [trackingOptions disableCity];
    }
    if([options containsObject:@"disableCountry"]){
        [trackingOptions disableCountry];
    }
    if([options containsObject:@"disableDeviceManufacturer"]){
        [trackingOptions disableDeviceManufacturer];
    }
    if([options containsObject:@"disableDeviceModel"]){
        [trackingOptions disableDeviceModel];
    }
    if([options containsObject:@"disableDMA"]){
        [trackingOptions disableDMA];
    }
    if([options containsObject:@"disableIDFA"]){
        [trackingOptions disableIDFA];
    }
    if([options containsObject:@"disableIDFV"]){
        [trackingOptions disableIDFV];
    }
    if([options containsObject:@"disableIPAddress"]){
        [trackingOptions disableIPAddress];
    }
    if([options containsObject:@"disableLanguage"]){
        [trackingOptions disableLanguage];
    }
    if([options containsObject:@"disableLatLng"]){
        [trackingOptions disableLatLng];
    }
    if([options containsObject:@"disableOSName"]){
        [trackingOptions disableOSName];
    }
    if([options containsObject:@"disableOSVersion"]){
        [trackingOptions disableOSVersion];
    }
    if([options containsObject:@"disablePlatform"]){
        [trackingOptions disablePlatform];
    }
    if([options containsObject:@"disableRegion"]){
        [trackingOptions disableOSVersion];
    }
    if([options containsObject:@"disableVersionName"]){
        [trackingOptions disablePlatform];
    }
    [[Amplitude instance] setTrackingOptions:trackingOptions];
}

RCT_EXPORT_METHOD(setUserPropertyForOnce:(NSString *)key value:(NSString *)value){
    AMPIdentify *identify = [[AMPIdentify identify] setOnce:key value:value];
    [[Amplitude instance] identify:identify];
}

RCT_EXPORT_METHOD(unSetUserProperty:(NSString *)key){
    AMPIdentify *identify = [[AMPIdentify identify] unset:key];
    [[Amplitude instance] identify:identify];
}

RCT_EXPORT_METHOD(logEvent:(NSString *) eventName params:(NSDictionary *)params){
    [[Amplitude instance] logEvent:eventName withEventProperties:params];
}
- (NSArray<NSString *> *)supportedEvents
{
    return @[];
}


@end
